packages/parser/src/types.ts

```diff
@@ -17,14 +19,14 @@ export interface ParserOptions {
      * will be able to see which paths in the schema contain circular references.
      *
      */
-    circular?: $RefParserOptions['dereference']['circular'];
+    circular?: boolean | 'ignore';
 
     /**
      * Callback invoked during circular reference detection.
      *
      * @param path - The path that is circular (ie. the `$ref` string)
      */
-    onCircular?: $RefParserOptions['dereference']['onCircular'];
```

Comment:
I tried to refactor this to use the `ParserOptions` export that `json-schema-ref-parser` offers, and not using the default from the dist, but the `DeepPartial` type that hey have wrapping `ParserOptions` blows away the types on functions and turns them into `{}`. 

Spent some time today trying to resolve this so I could submit a PR but threw in the towel because I couldn't get it working right.